src/main.rs

```diff
@@ -221,6 +200,13 @@ fn construct_config(matches: clap::ArgMatches, pattern_regex: &str) -> Result<Co
         .map_or_else(filesystem::default_path_separator, |s| Some(s.to_owned()));
     check_path_separator_length(path_separator.as_deref())?;
 
+    // Search full path if the pattern contains a path separator, even if the flag was not set.
+    // But only do so for hand-written patterns.
+    let search_full_path = matches.is_present("full-path")
+        || (cfg!(unix)
+            && !matches.is_present("glob")
+            && pattern_regex.contains(std::path::MAIN_SEPARATOR));
```

Comment:
On windows MAIN_SEPARATOR is a backslash "\", which is also a special character in regex, which means any regex that uses "\" will be forced to use full-path matching.

Reply:
Understood; will have to think through this.